packages/back-end/src/services/clickhouse.ts

```diff
@@ -288,3 +349,93 @@ WITH FILL
     bandwidth: parseInt(d.bandwidth) || 0,
   }));
 }
+
+export async function updateMaterializedColumns({
+  datasource,
+  columnsToAdd,
+  columnsToDelete,
+  columnsToRename,
+  finalColumns,
+  originalColumns,
+}: {
+  datasource: GrowthbookClickhouseDataSource;
+  columnsToAdd: MaterializedColumn[];
+  columnsToDelete: string[];
+  columnsToRename: { from: string; to: string }[];
+  finalColumns: MaterializedColumn[];
+  originalColumns: MaterializedColumn[];
+}) {
+  const client = createAdminClickhouseClient();
+
+  const orgId = datasource.organization;
+  const database = clickhouseUserId(orgId, datasource.id);
+  const eventsTableName = `${database}.events`;
+  const eventsViewName = `${database}.events_mv`;
+
+  // Track which columns the view should be recreated with in case of an error
+  let viewColumns = originalColumns;
+
+  logger.info(`Updating materialized columns; dropping view ${eventsViewName}`);
+  await client.command({ query: `DROP VIEW IF EXISTS ${eventsViewName}` });
+
+  let err = undefined;
+  try {
+    const addClauses = columnsToAdd
+      .map(
+        ({ columnName, datatype }) =>
+          `ADD COLUMN IF NOT EXISTS ${columnName} ${getClickhouseDatatype(
```

Comment:
Should we escape the column names in all of these queries?  Just thinking about if someone uses a SQL reserved keyword.  Because we don't allow spaces or punctuation, the damage someone could do is pretty limited, so maybe not a big deal.